File: Codewars/8KYU_Swift_JavaScript/17_sum_mixed_array/mixed_array.swift
/*
Given an array of integers as strings and numbers, return the sum of the array values as if all were numbers.

Return your answer as a number.
*/
import Foundation

func sumMix(_ arr: [Any]) -> Int {
  return arr
          .compactMap { Int("\($0)")} // compactMap filtered nil values
          .reduce(0,+)
}

print(sumMix(["1", 2, "3"]))

File: Codewars/8KYU_Swift_JavaScript/17_sum_mixed_array/mixed_array.js
/*
Given an array of integers as strings and numbers, return the sum of the array values as if all were numbers.

Return your answer as a number.
*/

function sumMix(x){
    return x
        .map(a => parseInt(a))  
        .reduce((a, b) => a + b);
}

console.log(sumMix([1, "2", "3"]));

/*
How to convert String -> Int:
    function parses: parseInt(a)
    plus operator: +a
    number constructor: Number(a)
*/